/*
 * Copyright (c) 2016 Cisco and/or its affiliates, and 
 *                    Cable Television Laboratories, Inc. ("CableLabs")
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

 /*
  * File has been generated by proto_generator.py
  */

syntax = "proto2";

message t_SsdController {
    enum t_ssdAction {
        SSD_STATUS = 1;
        SSD_START = 2;
        SSD_ABORT = 3;
    }
    enum t_ssdTransport {
        SSD_TRANSPORT_TFTP = 1;
        SSD_TRANSPORT_HTTP = 2;
    }
    enum t_ssdResult {
        SSD_RESULT_NONE = 0;
        SSD_RESULT_OK = 1;
        SSD_RESULT_GENERAL_ERROR = 2;
    }
    required t_ssdAction action = 1;
    required bytes server = 2;
    required bytes file = 3;
    required t_ssdTransport transport = 4;
    optional bytes manufacturerCvc = 5;
    optional bytes cosignerCvc = 6;
    optional uint32 status = 7;
    optional t_ssdResult result = 8;
}
message t_HalApi {
    optional t_SsdController ssdController = 1;
}
